File: python/problem_5.py
#
#2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
#
#What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
import math

res = 2520
for i in range(11, 21):
	res *= int(i / math.gcd(i, res))
print('LCM is ',res)

File: C++/problem_5.cpp

#include <iostream>

using namespace std;

 int LCM(int n1, int n2) {
		int lcm = 0;
		for (int i = 1; i < n2; i++) {
			lcm = (n1 > i) ? n1 : i;
			 for (lcm = (n1 > i) ? n1 : i; lcm <= (n1*n2); lcm++)
			 {
				 if(lcm % n1 == 0 && lcm % i == 0)
				 {
				   n1 = lcm;
		           break;
			 }
				 lcm += 1;
		       }
		}  
	   return lcm;
	 }
	 
int main()
{
    int res=LCM(20,100);
    cout<<res;
//output will be 1163962800
    return 0;
}
